### PR TITLE
Set certain upstream GN flags correctly in args.gn

### PR DESCRIPTION
**Pull request by niranjanyardi**
b/377271824
b/378934120

Set GN flags correctly by overriding them in args.gn rather than overriding it upstream code.

### CODE REVIEW

**Comment by niranjanyardi** [comment]
These flags are added to starboard toolchain args - so removing thesxe from upstream should have just "worked". It seems like I need to add this to args.gn for linux wdyt @andrewsavage1 ?

**Comment by niranjanyardi** [comment]
although these args are set in the starboard toolchain: https://source.corp.google.com/h/github/youtube/cobalt/+/main:starboard/linux/x64x11/toolchain/BUILD.gn;l=24
 the builds are failing .

I'll move them to args.gn wdyt @andrewsavage1 ?

**Comment by andrewsavage1** [comment]
> I'll move them to args.gn wdyt @andrewsavage1 ?

That's fine, do we have a follow up to clean up the code that emits these warnings (esp in //starboard/common)?

**Comment by niranjanyardi** [comment]
> > I'll move them to args.gn wdyt @andrewsavage1 ?
> 
> That's fine, do we have a follow up to clean up the code that emits these warnings (esp in //starboard/common)?

done, added bug

File: build/config/clang/clang.gni [hunk 1]
```diff
@@ -13,11 +13,11 @@ declare_args() {
   # Chrome OS's.
   clang_use_chrome_plugins =
       is_clang && !is_nacl && current_os != "zos" &&
-      default_toolchain != "//build/toolchain/cros:target" && !is_cobalt
+      default_toolchain != "//build/toolchain/cros:target"
```
Comment (kaidokert) [inline build/config/clang/clang.gni:16]: Just asking: this is _removing_ customizations vs upstream, right ?

File: cobalt/build/configs/android-arm/args.gn [hunk 2]
```diff
@@ -1,3 +1,7 @@
 target_os = "android"
 target_cpu = "arm"
+
+# TODO(b/378934120): Fix starboard compiler/linker errors while building with chromium flags
+clang_use_chrome_plugins = false
+enable_check_raw_ptr_fields = false
```
Comment (kaidokert) [inline cobalt/build/configs/android-arm/args.gn:6]: I'd recommend leaving a one-liner comment that explains where this is declared and what does it do. It seems it enables a number of checks from a custom Chromium bundled FindBadConstructs plugin ( here: https://chromium.googlesource.com/chromium/src/tools/clang/+/refs/heads/main/plugins )

**Review by kaidokert — APPROVED** [review]

**Review by oxve — APPROVED** [review]

File: cobalt/build/configs/android-arm/args.gn (hunk 2 above)
Comment (niranjanyardi) [reply cobalt/build/configs/android-arm/args.gn:6]: I'm just adding where this flag comes from , people can look up the definition in the original file.

**Review by andrewsavage1 — APPROVED** [review]

**Comment by niranjanyardi** [comment]
FYI: Don't force push this. I'll submit this alongwith gn.py documentation: cl/696670542  to ensure dev xp is smooth.  Submitting this change will force people to run gn.py or add these variables to their gn.py - and that is somewhat non-intuitive IMO

**Pull request merged**